provisioning/device/samples/solutions/BestPracticeSampleX509/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using CommandLine;
+using Microsoft.Azure.Devices.Logging;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Samples
+{
+    internal class Program
+    {
+        private const string SdkEventProviderPrefix = "Microsoft-Azure-";
+
+        public static async Task<int> Main(string[] args)
+        {
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // Set up logging
+            using ILoggerFactory loggerFactory = new LoggerFactory();
+            loggerFactory.AddColorConsoleLogger(
+                new ColorConsoleLoggerConfiguration
+                {
+                    // The SDK logs are written at Trace level. Set this to LogLevel.Trace to get ALL logs.
+                    MinLogLevel = LogLevel.Debug,
+                });
+            ILogger<Program> logger = loggerFactory.CreateLogger<Program>();
+
+            // Instantiating this seems to do all we need for outputting SDK events to our console log.
+            using var sdkLogs = new ConsoleEventListener(SdkEventProviderPrefix, logger);
+
+            // Run the sample
+            TimeSpan runningTime = parameters.ApplicationRunningTime != null
+                ? TimeSpan.FromSeconds((double)parameters.ApplicationRunningTime)
+                : Timeout.InfiniteTimeSpan;
+
+            try
+            {
+                var sample = new BestPracticeSampleX509(parameters.GlobalDeviceEndpoint, parameters.IdScope, parameters.CertificateName, parameters.CertificatePassword, parameters.TransportType, logger, parameters);
+                await sample.RunSampleAsync(runningTime);
+            }
+            catch (OperationCanceledException)
+            { logger.LogDebug("OperationCanceledException caught."); }
+            catch (Exception ex)
+            { logger.LogDebug($"Exception caught. \n{ex}"); }
```

Comment:
Observe standard bracing.

```suggestion
            catch (OperationCanceledException)
            {
                logger.LogDebug("OperationCanceledException caught.");
            }
            catch (Exception ex)
            {
                logger.LogDebug($"Exception caught. \n{ex}");
            }
```